zerver/migrations/0470_set_default_value_for_create_multiuse_invite_group.py

```diff
@@ -0,0 +1,57 @@
+# Generated by Django 4.2.1 on 2023-06-03 10:53
+
+from django.db import migrations
+from django.db.backends.base.schema import BaseDatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+
+
+def set_default_value_for_create_multiuse_invite_group(
+    apps: StateApps, schema_editor: BaseDatabaseSchemaEditor
+) -> None:
+    Realm = apps.get_model("zerver", "Realm")
+    UserGroup = apps.get_model("zerver", "UserGroup")
+
+    Realm.POLICY_NOBODY = 6
+    Realm.POLICY_ADMINS_ONLY = 2
+
+    UserGroup.NOBODY_GROUP_NAME = "role:nobody"
+    UserGroup.ADMINISTRATORS_GROUP_NAME = "role:administrators"
+    default_group_name = "role:moderators"
+
+    for realm in Realm.objects.all():
+        if realm.create_multiuse_invite_group is None:
+            if realm.invite_to_realm_policy == Realm.POLICY_NOBODY:
+                nobody_group = UserGroup.objects.get(
+                    name=UserGroup.NOBODY_GROUP_NAME, realm=realm, is_system_group=True
+                )
+                realm.create_multiuse_invite_group = nobody_group
+
+            elif realm.invite_to_realm_policy == Realm.POLICY_ADMINS_ONLY:
+                admins_group = UserGroup.objects.get(
+                    name=UserGroup.ADMINISTRATORS_GROUP_NAME, realm=realm, is_system_group=True
+                )
+                realm.create_multiuse_invite_group = admins_group
+
+            else:
+                default_group = UserGroup.objects.get(
+                    name=default_group_name, realm=realm, is_system_group=True
+                )
+                realm.create_multiuse_invite_group = default_group
```

Comment:
So I think we actually want the default for existing administrators that hadn't specified `NOBODY` as their policy to be `role:administrators`, because that way we don't adjust permissions for anyone upwards unexpectedly.

Reply:
``` diff
diff --git a/zerver/migrations/0470_set_default_value_for_create_multiuse_invite_group.py b/zerver/migrations/0470_set_default_value_for_create_multiuse_invite_group.py
index f162fb3e05..6a82d6afea 100644
--- a/zerver/migrations/0470_set_default_value_for_create_multiuse_invite_group.py
+++ b/zerver/migrations/0470_set_default_value_for_create_multiuse_invite_group.py
@@ -11,34 +11,23 @@ def set_default_value_for_create_multiuse_invite_group(
     Realm = apps.get_model("zerver", "Realm")
     UserGroup = apps.get_model("zerver", "UserGroup")
 
-    Realm.POLICY_NOBODY = 6
-    Realm.POLICY_ADMINS_ONLY = 2
-
-    UserGroup.NOBODY_GROUP_NAME = "role:nobody"
     UserGroup.ADMINISTRATORS_GROUP_NAME = "role:administrators"
-    default_group_name = "role:moderators"
 
     for realm in Realm.objects.all():
-        if realm.create_multiuse_invite_group is None:
-            if realm.invite_to_realm_policy == Realm.POLICY_NOBODY:
-                nobody_group = UserGroup.objects.get(
-                    name=UserGroup.NOBODY_GROUP_NAME, realm=realm, is_system_group=True
-                )
-                realm.create_multiuse_invite_group = nobody_group
-
-            elif realm.invite_to_realm_policy == Realm.POLICY_ADMINS_ONLY:
-                admins_group = UserGroup.objects.get(
-                    name=UserGroup.ADMINISTRATORS_GROUP_NAME, realm=realm, is_system_group=True
-                )
-                realm.create_multiuse_invite_group = admins_group
-
-            else:
-                default_group = UserGroup.objects.get(
-                    name=default_group_name, realm=realm, is_system_group=True
-                )
-                realm.create_multiuse_invite_group = default_group
-
-            realm.save(update_fields=["create_multiuse_invite_group"])
+        if realm.create_multiuse_invite_group is not None:
+            continue
+
+        # Prior to the new create_multiuse_invite_group field being
+        # created, multi-use invitation links could only be created
+        # and managed by administrators, regardless of
+        # invite_to_realm_policy. We replicate that policy for the
+        # initial value of the new setting.
+        admins_group = UserGroup.objects.get(
+            name=UserGroup.ADMINISTRATORS_GROUP_NAME, realm=realm, is_system_group=True
+        )
+        realm.create_multiuse_invite_group = admins_group
+
+        realm.save(update_fields=["create_multiuse_invite_group"])
 
 
 class Migration(migrations.Migration):
```